clients/src/main/java/org/apache/kafka/clients/admin/CreateDelegationTokenOptions.java

```diff
@@ -31,10 +32,10 @@
 @InterfaceStability.Evolving
 public class CreateDelegationTokenOptions extends AbstractOptions<CreateDelegationTokenOptions> {
     private long maxLifeTimeMs = -1;
-    private List<KafkaPrincipal> renewers =  new LinkedList<>();
+    private List<KafkaPrincipal> renewers = Collections.emptyList();
```

Comment:
There may be an assumption that the collection will be mutable because it is passed around and used elsewhere -- potentially being mutated.  It feels to me that it would be better to embrace immutability here and wherever else such a collection is used.  This ticket (`"Modify CreateDelegationTokenOptions Renewers Implementation"`) would be closed in favor of one that has a broader scope (e.g. `"Embrace immutability in delegation token processing"`).  This would also require a KIP since the public API would be changing (maybe not in signature, but definitely in semantics).